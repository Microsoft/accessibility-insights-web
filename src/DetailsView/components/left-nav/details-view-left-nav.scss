// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../../common/styles/colors.scss';
@import '../../../common/styles/common.scss';

:global(.reflow-ui) {
    .left-nav {
        :global {
            .ms-Nav-compositeLink {
                &.is-selected {
                    background-color: $communication-tint-20;
                    a {
                        background-color: $communication-tint-20 !important;
                    }
                }
                &.is-expanded,
                &.is-expanded + ul {
                    background-color: $communication-tint-40;
                    .index-circle {
                        color: $neutral-0;
                        background: $index-circle-background;
                        border-color: $neutral-0;
                    }
                }
                &:hover {
                    a {
                        background-color: $nav-link-hover !important;
                    }
                }
            }
        }
    }
}

:global(.details-view-body-nav-content-layout) {
    .left-nav {
        max-height: calc(100vh - (#{$detailsViewTotalHeaderHeight}));
        height: calc(100vh - (#{$detailsViewTotalHeaderHeight}));
    }
}

.left-nav {
    overflow-y: auto;
    border-right: 1px solid $neutral-8;
    z-index: 100;
    background-color: $neutral-0;
    :global {
        .dark-gray {
            color: $neutral-60;
        }
        .is-selected .button-flex-container {
            color: $always-black;
        }
        .button-flex-container {
            padding-left: 5%;
            padding-right: 8%;
            width: $detailsViewLeftNavWidth;
            color: $primary-text;
            .status-icon {
                font-size: 24px;
            }
        }
        .ms-Nav-link {
            height: fit-content;
        }
        .overview-label {
            padding-left: 10%;
            text-align: left;
            margin-top: 6px;
            margin-bottom: 6px;
            display: flex;
            flex-direction: column;
        }
        .overview-name {
            line-height: 20px;
        }
        .overview-percent {
            color: $neutral-55;
            line-height: 16px;
            font-size: 12px;
        }
        .index-circle {
            border: 1px solid;
            display: inline-block;
            border-radius: 50%;
            border: 1px solid;
            width: 1.5em;
            line-height: 1.5em;
            padding: 0.07em;
            margin-top: 0.23em;
        }
        .test-name {
            padding-left: 10%;
            display: inline-block;
        }
        .ms-Nav-navItems {
            margin-top: 0px !important;
        }
        .ms-Nav-compositeLink {
            .ms-Button-label {
                color: $neutral-100;
            }
            .button-flex-container {
                display: flex;
                flex-wrap: nowrap;
                justify-content: flex-start;
                height: 100%;
                align-items: center;
            }
            .index-circle {
                color: $secondary-text;
                background: $neutral-0;
                border-color: $neutral-20;
            }
            &.is-selected {
                a {
                    background-color: $communication-tint-40;
                }
                a::after {
                    border-left: 0px;
                }
                .ms-Button-label {
                    color: $always-black;
                }
                .index-circle {
                    color: $neutral-0;
                    background: $index-circle-background;
                    border-color: $neutral-0;
                }
            }
        }
        .ms-Pivot-text {
            font-size: $fontSizeS;
        }
        .ms-Pivot-icon {
            font-size: 9px;
        }
        button {
            text-align: left;
        }
    }
}
