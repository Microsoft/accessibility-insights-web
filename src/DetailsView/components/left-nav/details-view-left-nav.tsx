// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { mapValues } from 'lodash';
import * as React from 'react';

import { NamedSFC } from '../../../common/react/named-sfc';
import { AssessmentStoreData } from '../../../common/types/store-data/assessment-result-data';
import { FeatureFlagStoreData } from '../../../common/types/store-data/feature-flag-store-data';
import { VisualizationType } from '../../../common/types/visualization-type';
import { DetailsRightPanelConfiguration } from '../details-view-right-panel';
import { DetailsViewSwitcherNavConfiguration, LeftNavDeps } from '../details-view-switcher-nav';

export type DetailsViewLeftNavDeps = {
    assessmentsProvider: AssessmentsProvider;
} & LeftNavDeps;

export type DetailsViewLeftNavProps = {
    deps: DetailsViewLeftNavDeps;
    selectedTest: VisualizationType;
    switcherNavConfiguration: DetailsViewSwitcherNavConfiguration;
    rightPanelConfiguration: DetailsRightPanelConfiguration;
    featureFlagStoreData: FeatureFlagStoreData;
    assessmentStoreData: AssessmentStoreData;
};

export const DetailsViewLeftNav = NamedSFC<DetailsViewLeftNavProps>('DetailsViewLeftNav', props => {
    const { deps, selectedTest, switcherNavConfiguration, rightPanelConfiguration, featureFlagStoreData, assessmentStoreData } = props;

    const { assessmentsProvider } = deps;

    const selectedKey: string = rightPanelConfiguration.GetLeftNavSelectedKey({ visualizationType: selectedTest });

    const leftNav: JSX.Element = (
        <div className="left-nav main-nav">
            <switcherNavConfiguration.LeftNav
                {...props}
                assessmentsProvider={assessmentsProvider}
                selectedKey={selectedKey}
                assessmentsData={mapValues(assessmentStoreData.assessments, data => data.testStepStatus)}
            />
        </div>
    );

    return leftNav;
});
