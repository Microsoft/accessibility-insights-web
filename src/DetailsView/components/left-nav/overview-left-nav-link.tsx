// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { css } from '@uifabric/utilities';
import { Icon } from 'office-ui-fabric-react';
import * as React from 'react';

import { NamedFC } from '../../../common/react/named-fc';
import { BaseLeftNavLinkProps } from '../base-left-nav';

export const OverviewLeftNavLink = NamedFC<Pick<BaseLeftNavLinkProps, 'link'>>(
    'OverviewLeftNavLink',
    ({ link }) => {
        return (
            <span className={'button-flex-container'} aria-hidden="true">
                <span>
                    <Icon iconName="home" className={css('status-icon', 'dark-gray')} />
                </span>
                <span className="ms-Button-label overview-label">
                    <span className="overview-name">{link.name}</span>
                    <span className="overview-percent">{link.percentComplete}% Completed</span>
                </span>
            </span>
        );
    },
);
