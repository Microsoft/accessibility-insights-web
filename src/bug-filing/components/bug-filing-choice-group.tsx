// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
import * as React from 'react';

import { UserConfigMessageCreator } from '../../common/message-creators/user-config-message-creator';
import { NamedSFC } from '../../common/react/named-sfc';
import { BugFilingService } from '../types/bug-filing-service';

export type BugFilingChoiceGroupProps = {
    deps: BugFilingChoiceGroupDeps;
    selectedBugFilingService: BugFilingService;
    bugFilingServices: BugFilingService[];
};

export type BugFilingChoiceGroupDeps = {
    userConfigMessageCreator: UserConfigMessageCreator;
};

export const BugFilingChoiceGroup = NamedSFC<BugFilingChoiceGroupProps>('BugFilingChoiceGroup', props => {
    const { userConfigMessageCreator } = props.deps;

    const getOptions: () => IChoiceGroupOption[] = () => {
        return props.bugFilingServices.map(service => {
            return {
                key: service.key,
                text: service.displayName,
            };
        });
    };

    const onChange = (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, option?: IChoiceGroupOption) => {
        userConfigMessageCreator.setBugService(option.key);
    };

    return (
        <ChoiceGroup
            className={'bug-filing-choice-group'}
            onChange={onChange}
            options={getOptions()}
            selectedKey={props.selectedBugFilingService.key}
        />
    );
});
