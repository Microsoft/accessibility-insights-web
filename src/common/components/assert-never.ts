// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// This function helps with exhaustiveness checking in switch statements
// If used in the default case, TS will throw an error unless all cases are handled.
export const assertNever = (x: never): never => {
    throw Error(
        'This error should never be thrown when strict null checks are applied to this project',
    );
};
