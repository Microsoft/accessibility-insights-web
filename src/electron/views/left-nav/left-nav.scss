// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/common.scss';

.left-nav {
    display: flex;
    flex-direction: column;
    width: 249px;
    border-right: 1px solid $neutral-alpha-8;
    height: 100%;

    .header-container {
        display: flex;
        align-items: center;
        height: 40px;
        border-bottom: 1px solid $neutral-alpha-8;
    }
}
