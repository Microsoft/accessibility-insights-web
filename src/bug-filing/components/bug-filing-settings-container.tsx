// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { BugServiceProperties } from '../../common/types/store-data/user-configuration-store';
import { SettingsDeps } from '../../DetailsView/components/settings-panel/settings/settings-props';
import { BugFilingServiceProvider } from '../bug-filing-service-provider';
import { BugFilingService } from '../types/bug-filing-service';
import { BugFilingChoiceGroup, BugFilingChoiceGroupDeps } from './bug-filing-choice-group';

export interface BugFilingSettingsContainerProps {
    deps: BugFilingSettingsContainerDeps;
    selectedBugFilingService: BugFilingService;
    selectedBugFilingServiceData: BugServiceProperties;
}

export type BugFilingSettingsContainerDeps = {
    bugServiceProvider: BugFilingServiceProvider;
} & BugFilingChoiceGroupDeps &
    SettingsDeps;

export const BugFilingSettingsContainer = NamedSFC<BugFilingSettingsContainerProps>('BugFilingSettingsContainer', props => {
    const { deps, selectedBugFilingService, selectedBugFilingServiceData } = props;
    const SettingsForm = selectedBugFilingService.renderSettingsForm;
    const bugFilingServices = deps.bugServiceProvider.allVisible();

    return (
        <>
            <BugFilingChoiceGroup deps={deps} selectedBugFilingService={selectedBugFilingService} bugFilingServices={bugFilingServices} />
            <SettingsForm deps={deps} settings={selectedBugFilingServiceData} />
        </>
    );
});
