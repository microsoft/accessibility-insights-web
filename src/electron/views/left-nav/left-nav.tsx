// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { Icon } from '@fluentui/react';
import { NamedFC } from 'common/react/named-fc';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { BaseLeftNav, BaseLeftNavLink } from 'DetailsView/components/base-left-nav';
import { NavLinkRenderer } from 'DetailsView/components/left-nav/nav-link-renderer';
import { LeftNavItem } from 'electron/types/left-nav-item';
import { LeftNavItemKey } from 'electron/types/left-nav-item-key';
import * as styles from 'electron/views/left-nav/left-nav.scss';
import * as React from 'react';

export type LeftNavDeps = {
    leftNavItems: LeftNavItem[];
    navLinkRenderer: NavLinkRenderer;
};

export type LeftNavProps = {
    deps: LeftNavDeps;
    selectedKey: LeftNavItemKey;
    featureFlagStoreData: FeatureFlagStoreData;
};

export const leftNavAutomationId = 'left-nav';

export const LeftNav = NamedFC<LeftNavProps>('LeftNav', props => {
    const { deps } = props;
    const leftNavItemsFiltered = deps.leftNavItems.filter(item =>
        filterForFeatureFlag(item, props.featureFlagStoreData),
    );
    const leftLinkItems: BaseLeftNavLink[] = leftNavItemsFiltered.map((item, index) => ({
        name: item.displayName,
        key: item.key,
        onClickNavLink: item.onSelect,
        onRenderNavLink: deps.navLinkRenderer.renderVisualizationLink,
        url: '',
        index: index + 1,
        iconProps: {
            className: 'hidden',
        },
        forceAnchor: true,
    }));

    return (
        <div data-automation-id={leftNavAutomationId} className={styles.leftNav}>
            <div className={styles.headerContainer}>
                <Icon iconName={'Rocket'} className={styles.rocketIcon} />
                <h3>FastPass</h3>
            </div>
            <BaseLeftNav selectedKey={props.selectedKey} links={leftLinkItems} />
        </div>
    );
});

const filterForFeatureFlag = (
    config: LeftNavItem,
    featureFlagStoreData: FeatureFlagStoreData,
): boolean => {
    return config.featureFlag === undefined || featureFlagStoreData[config.featureFlag];
};
