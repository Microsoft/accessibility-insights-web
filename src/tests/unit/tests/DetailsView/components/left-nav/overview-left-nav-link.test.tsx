// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import { LeftNavLinkProps, NavLinkForLeftNav } from '../../../../../../DetailsView/components/details-view-left-nav';
import { OverviewLeftNavLink } from '../../../../../../DetailsView/components/left-nav/overview-left-nav-link';

describe('OverviewLeftNavLink', () => {
    it('renders', () => {
        const props: LeftNavLinkProps = {
            link: {
                name: 'test-props-link-name',
                percentComplete: 42,
            } as NavLinkForLeftNav,
            renderIcon: undefined,
        };

        const wrapper = shallow(<OverviewLeftNavLink {...props} />);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
