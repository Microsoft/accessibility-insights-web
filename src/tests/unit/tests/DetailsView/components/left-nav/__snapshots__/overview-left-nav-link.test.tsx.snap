// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewLeftNavLink renders 1`] = `
<div
  aria-hidden="true"
  className="button-flex-container"
>
  <div>
    <StyledIconBase
      className="status-icon dark-gray"
      iconName="home"
    />
  </div>
  <div
    className="ms-Button-label overview-label"
  >
    <div
      className="overview-name"
    >
      test-props-link-name
    </div>
    <div
      className="overview-percent"
    >
      42
      % Completed
    </div>
  </div>
</div>
`;
